src/spac/data_utils.py

```diff
@@ -502,44 +503,73 @@ def downsample_cells(data, annotation, n_samples=None,
     Examples
     --------
     >>> df = pd.DataFrame({
-    ...    'annotation': ['a', 'a', 'a', 'b', 'b', 'c'],
+    ...    'annotation1': ['a', 'a', 'b', 'b', 'c', 'c'],
+    ...    'annotation2': ['x', 'y', 'x', 'y', 'x', 'y'],
     ...    'value': [1, 2, 3, 4, 5, 6]
     ... })
-    >>> print(downsample_cells(df, 'annotation', n_samples=2))
-    """
-    # Check if the column to downsample on exists
-    if annotation not in data.columns:
-        raise ValueError(
-            f"Column {annotation} does not exist in the dataframe"
+    >>> print(
+            downsample_cells(df, ['annotation1', 'annotation2'], n_samples=2)
         )
+    """
+
+    # Convert annotation to list if it's a string
+    if isinstance(annotation, str):
+        annotation = [annotation]
+
+    # Check if the columns to downsample on exists
+    for col in annotation:
+        if col not in data.columns:
+            raise ValueError(f"Column {col} does not exist in the dataframe")
 
     if n_samples is not None:
+        # Combine annotations into a single column if there
+        # are multiple annotations
+        if len(annotation) > 1:
+            combined_col_name = '_combined_'
+            data[combined_col_name] = data[annotation].apply(
+                lambda row: '_'.join(row.values.astype(str)), axis=1)
+            grouping_col = combined_col_name
+        else:
+            grouping_col = annotation[0]
+
         # Stratify selection
         if stratify:
             # Determine frequencies of each group
-            freqs = data[annotation].value_counts(normalize=True)
+            freqs = data[grouping_col].value_counts(normalize=True)
             n_samples_per_group = (freqs * n_samples).astype(int)
+
+            # Increase the number of samples for groups with 0
+            # calculated samples.
+            n_samples_per_group = n_samples_per_group.apply(

```

Comment:
@fangliu117 , would you add a unit test for this case (group that has zero samples?)

Reply:
Yes, added.